client/src/pages/Homepage.js

```diff
@@ -62,12 +46,16 @@ const Homepage = () => {
     };
     const response = await fetch(API_URL + "/habit", config);
     const data = await response.json();
+    const habitIds = user.habits;
+    setMyHabits(data.filter((habit) => habitIds.includes(habit._id)));
     setHabits(data);
   };
```

Comment:
since `getAllHabits` is responsible for multiple things now, let's give it a new name. this is what I would do to improve readability:
- rename `getAllHabits` function to `syncHabits`.
- rename `data` variable to `allHabits`
- rename `habitIds` variable to `myHabitIds`

Reply:
alright, I'll add these things.
@lugenx  I was wondering if there is  a way to reload the userContext when we add/remove an habit?

Reply:
The habits are not loaded when we logout, refresh the page and login because of token I guess, 
* I see that when we logout, refresh the page and login, the token is not generated or is still null because of which the fetchUserData function is not running as a result we don't get to see allhabits as well as user specific habits that depend on userContext for data.
* When we reload the page the token is present and everything works fine.

I think this is the problem, not sure how to deal with it.

Reply:
Yes!! Thank you @victorchrollo14, you found the issue. We fetch user data inside the `userContext`, and that doesn't run right after login. 

lets make this work:

1. Inside the `services/auth.js` please make login function to return user data too, so we can get token from that. in login function make the below changes:

```javascript

// Remove
    return response.status; 

// Add
    return {
      status: response.status,
      data: json,
    }; 
```

2. inside `components/Login.js`
- Import `userContext` at the top.
```javascript
import { useUserContext } from "../contexts/UserContext.js"; 
```
- then get `setToken` from context
```javascript
 const { setToken } = useUserContext();
```
- then inside `handleSubmit`, let's set the token inside the userContext so it fetches user data right after login

`try` block inside `handleSubmit` should look like this:

```javascript
    try {
      // const responseStatus = await login(loginData); // <-- remove this
      const response = await login(loginData); // <-- add this
      const responseStatus = response.status; // <--add this
      const token = await response.data.token; // <-- add this

      if (responseStatus === 200) {
        setLoggedIn(true);
        setToken(token); // <-- add this
        navigate("/");
      } else if (responseStatus === 403) {
        setLoginFailMessage("User does not exist");
      } else if (responseStatus === 400) {
        setLoginFailMessage("Username or password is incorrect");
      }
    }
```

Reply:
[Screencast from 25-08-23 10:55:28 AM IST.webm](https://github.com/lugenx/ecohabit/assets/89346667/b9dba251-84a6-4166-8944-4c549bd00819)